bin/geninfo

```diff
@@ -2296,15 +2337,18 @@ sub intermediate_json_to_info($$$)
 
 			$branch_num = 0;
 			# Branch data
-			if ($br_coverage && !$brexcl->{$line}) {
+			if ($br_coverage && (!defined($brexcl->{$line}) || ($brexcl->{$line} != 1))) {
 				for my $b (@$branches) {
 					my $brcount = $b->{"count"};
+					my $is_exception = $b->{"throw"};
 
-					if (!defined($brcount) || $unexec) {
-						$brcount = "-";
+					if (!$is_exception || ((!defined($brexcl->{$line}) || ($brexcl->{$line} != 2)) && ($br_coverage != 2))) {
+						if (!defined($brcount) || $unexec) {
+							$brcount = "-";
+						}
+						print($fd "BRDA:$line,0,$branch_num,".
+						      "$brcount\n");
```

Comment:
I left the branch_number to increase even if the branch was excluded from coverage for coherency with reports generated with exception coverage enabled. That way the non-exception branches will have the same "identification numbers" (just thinking of a case when two developers discuss over the net about coverage but they generated the reports with different options).

This will also avoid strange results when combining tracefiles generated with different options (that may be rare, but can happen).